src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -94,7 +94,7 @@ private void CheckIfAssignedDuringLocalFunctionReplay(Symbol symbol, SyntaxNode
                         // Local functions can "call forward" to after a variable has
                         // been declared but before it has been assigned, so we can never
                         // consider the declaration location when reporting errors.
-                        ReportUnassigned(symbol, node, slot, skipIfUseBeforeDeclaration: false);
+                        ReportUnassignedIfOutsideLocalFunction(symbol, node, slot, skipIfUseBeforeDeclaration: false);
```

Comment:
No, you could call another local function (or even the same local function, recursively) from inside a local function.